src/hotspot/share/cds/heapShared.cpp

```diff
@@ -461,11 +461,13 @@ void HeapShared::archive_objects(ArchiveHeapInfo *heap_info) {
     // Cache for recording where the archived objects are copied to
     create_archived_object_cache();
 
-    log_info(cds)("Heap range = [" PTR_FORMAT " - "  PTR_FORMAT "]",
-                   UseCompressedOops ? p2i(CompressedOops::begin()) :
-                                       p2i((address)G1CollectedHeap::heap()->reserved().start()),
-                   UseCompressedOops ? p2i(CompressedOops::end()) :
-                                       p2i((address)G1CollectedHeap::heap()->reserved().end()));
+    if (UseCompressedOops || UseG1GC) {
+      log_info(cds)("Heap range = [" PTR_FORMAT " - "  PTR_FORMAT "]",
+                    UseCompressedOops ? p2i(CompressedOops::begin()) :
+                                        p2i((address)G1CollectedHeap::heap()->reserved().start()),
+                    UseCompressedOops ? p2i(CompressedOops::end()) :
+                                        p2i((address)G1CollectedHeap::heap()->reserved().end()));
```

Comment:
Does this work if you build a minimal VM with only SerialGC? I would expect the build to fail.

Reply:
Yes, I verified locally that this works on a minimal build

Reply:
Doh! Of course it does. The minimal VM doesn't have CDS either. Sorry.

If you build explicitly without G1 then I would expect this to fail.

Reply:
Not that I think we need to accommodate that. I suspect CDS already has some code dependencies on G1